features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/thresholding/FlowThresholding.java

```diff
@@ -344,25 +422,26 @@ private Session(final ThresholdingSession thresholdingSession,
             this.dataCollection = dataCollection;
 
             this.packageDefinition =  packageDefinition;
+            this.exporterIpAddress = exporterIpAddress;
         }
 
         public void process(final Instant now, final FlowDocument document) {
             if (document.getInputSnmp() != null &&
                 document.getInputSnmp() != 0 &&
                 (document.getDirection() == Direction.INGRESS || document.getDirection() == Direction.UNKNOWN)) {
-                final var applicationKey = new ApplicationKey(document.getInputSnmp(),
-                                                              Direction.INGRESS,
-                                                              document.getApplication());
-                this.applications.computeIfAbsent(applicationKey, k -> new AtomicLong(0)).addAndGet(document.getBytes());
+                final IndexKey indexKey = new IndexKey(document.getInputSnmp(), Direction.INGRESS);
+                synchronized (this.applications) {
```

Comment:
This really needs synchronisation. But a RWLock is better suited. And the writing part is missing locking.